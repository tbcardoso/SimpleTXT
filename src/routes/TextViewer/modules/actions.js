/*
 * action types
 */

export const SELECT_FILE_SUCCESS = 'SELECT_FILE_SUCCESS'
export const LOAD_FILE_SUCCESS = 'LOAD_FILE_SUCCESS'



/*
 * action creators
 */

export function selectFileSuccess(filePath) {
  return { type: SELECT_FILE_SUCCESS, filePath }
}

export function loadFileSuccess(filePath, fileContent) {
  return { type: LOAD_FILE_SUCCESS, filePath, fileContent }
}