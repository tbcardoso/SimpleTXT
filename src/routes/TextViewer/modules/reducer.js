import { combineReducers } from 'redux';
import * as Actions from './actions';
import about from '../../../assets/about';

const selectedFileInitialState = {
	filePath: '',
	fileContent: '',
	loadingFilePath: '',
	error: '',
};

function selectedFile(state = selectedFileInitialState, action) {
	switch (action.type) {
		case Actions.SELECT_FILE_SUCCESS:
			return { ...state, loadingFilePath: action.filePath };
		case Actions.LOAD_FILE_SUCCESS:
			return { ...state, loadingFilePath: '', filePath: action.filePath, fileContent: action.fileContent, error: '' };
		case Actions.LOAD_FILE_ERROR:
			return { ...state, loadingFilePath: '', error: action.error };
		case Actions.ABOUT_SCREEN_SELECTED:
			return { ...state, loadingFilePath: '', filePath: 'About', fileContent: about, error: '' };
		default:
			return state;
	}
}

export default combineReducers({
	selectedFile,
});
