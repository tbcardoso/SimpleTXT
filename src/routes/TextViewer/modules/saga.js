import { call, put, takeLatest } from 'redux-saga/effects';
import * as Actions from './actions';
import RNFetchBlob from 'react-native-fetch-blob';
import { ToastAndroid } from 'react-native';

function * selectedFileSaga(action) {
	console.log('Selected file Saga: ', action);

	try {
		const data = yield call(RNFetchBlob.fs.readFile, action.filePath, 'utf8');
		//console.log('File data: ', data);
		yield put(Actions.loadFileSuccess(action.filePath, data));
	} catch (e) {
		console.log('Load File Error: ', e);
		ToastAndroid.show(e.message, ToastAndroid.SHORT)
		yield put(Actions.loadFileError(action.filePath, e.message));
	}
}

export default function * saga() {
	yield takeLatest(Actions.SELECT_FILE_SUCCESS, selectedFileSaga);
}
