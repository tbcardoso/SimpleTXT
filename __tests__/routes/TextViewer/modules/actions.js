/* eslint-env jest */
import * as Actions from '../../../../src/routes/TextViewer/modules/actions';

describe('TextViewer Action Creators', () => {
	test('Action Creator loadStartupFile', () => {
		expect(Actions.loadStartupFile()).toMatchSnapshot();
	});

	test('Action Creator selectFileSuccess', () => {
		expect(Actions.selectFileSuccess('test/filepath.txt')).toMatchSnapshot();
	});

	test('Action Creator loadFileSuccess', () => {
		expect(Actions.loadFileSuccess('test/filepath.txt', 'Lorem ipsum...')).toMatchSnapshot();
	});

	test('Action Creator loadFileError', () => {
		expect(Actions.loadFileError('test/filepath.txt', 'Test error message.')).toMatchSnapshot();
	});

	test('Action Creator aboutScreenSelected', () => {
		expect(Actions.aboutScreenSelected()).toMatchSnapshot();
	});
});
