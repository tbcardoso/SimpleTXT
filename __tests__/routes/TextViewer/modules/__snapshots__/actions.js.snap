// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextViewer Action Creators Action Creator aboutScreenSelected 1`] = `
Object {
  "type": "ABOUT_SCREEN_SELECTED",
}
`;

exports[`TextViewer Action Creators Action Creator loadFileError 1`] = `
Object {
  "error": "Test error message.",
  "filePath": "test/filepath.txt",
  "type": "LOAD_FILE_ERROR",
}
`;

exports[`TextViewer Action Creators Action Creator loadFileSuccess 1`] = `
Object {
  "fileContent": "Lorem ipsum...",
  "filePath": "test/filepath.txt",
  "type": "LOAD_FILE_SUCCESS",
}
`;

exports[`TextViewer Action Creators Action Creator loadStartupFile 1`] = `
Object {
  "type": "LOAD_STARTUP_FILE",
}
`;

exports[`TextViewer Action Creators Action Creator selectFileSuccess 1`] = `
Object {
  "filePath": "test/filepath.txt",
  "type": "SELECT_FILE_SUCCESS",
}
`;
